### DISCUSSION TITLE
데이터가 변경되지 않는 이벤트에 대한 처리는 누가 담당해야할까요?

### DISCUSSION THREAD

**Discussion by byhhh2** [woowacourse-study/2022-code-review-study#15] [Lotto] [answered]
크루들의 코드에서 많이 갈리는 것같은 부분인데,
데이터가 변경되지 않는 이벤트 (토글을 통해 화면의 레이아웃을 바꾼다던지, 모달을 띄운다던지)에 대한 처리는 `View`가 담당하는게 맞을까요 `Controller`가 맞을까요?

- view가 담당

```js
// in view

toggle.addEventListener('click', () => this.changeLayout())

...

changeLayout() {
  // 레이아웃을 변경
}
```

- controller가 담당

```js
// in view

toggle.addEventListener('click', () => controller.toggle()) // controller에게 이벤트가 발생했음을 알림

...

changeLayout() {
  // 레이아웃을 변경
}

// in controller

toggle() {
  this.view.changeLayout() // view에게 layout를 변경하라고 시키는 형식
}
```

**Comment by wonsss** [comment, marked as answer]
저와 같은 고민을 하고 계셨군요!
`데이터가 변경되지 않는` 이벤트에 대한 처리는 view에서 끝내는 편이 나을 것 같아요.
그 이유를 알아보기 위해 반대의 경우부터 살펴보면, `데이터가 변경되는` 이벤트에 대한 처리는 model이 개입을 해야 하기 때문에, view는 model에 의존하는 controller를 통해 이벤트를 처리하고 있어요.
그런데 데이터가 변하지 않아서 model이 개입할 필요가 없는 경우라면, dom 요소에서 발생한 이벤트를 view에서 감지하고 처리까지 해주는 것이 깔끔한 것 같아요. 굳이 기존의 방법과 통일을 해야 한다는 이유로, 별다른 효용도 없이 번거롭게 controller를 거쳐서 다시 view의 메서드를 실행하는 것은 지나치지 않을까요?
다른 분들의 의견도 궁금합니다!
